community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/ast/rewriters/isolateAggregation.scala

```diff
@@ -92,6 +73,92 @@ case object isolateAggregation extends Rewriter {
       q.copy(clauses = newClauses)(q.position)
   }
 
+  private def createRewriterFor(withReturnItems: Set[ReturnItem]): Rewriter = {
+    def inner: Rewriter =
+      Rewriter.lift {
+        case original: Expression =>
+          val rewrittenExpression = withReturnItems.collectFirst {
+            case item@AliasedReturnItem(expression, variable) if original == expression =>
+              item.alias.get.copyId
+          }
+          rewrittenExpression getOrElse original
+      }
+
+    /*
+    Instead of using topDown, we do it manually here, because we don't want to rewrite the return aliases,
+    only the expressions
+    */
+    new Rewriter {
```

Comment:
Could this be refactored so that we can write unit tests for the rewriter. I think the rewriter is non-obvious enough to benefit from a couple of unit tests.